pages/search/search.jsx

```diff
@@ -17,32 +18,29 @@ export default React.createClass({
   componentDidMount() {
     this.setSearchQuery(this.props.location.query.keyword);
   },
-  setSearchQuery(searchQuery) {
-    if (searchQuery !== this.searchInput.value) {
-      this.searchInput.focus();
-      this.searchInput.value = searchQuery ? decodeURIComponent(searchQuery) : ``;
-      this.setState({searchQuery: searchQuery});
+  setSearchQuery(newSearchQuery) {
+    let prevSearchQuery = this.refs.searchInput.state.value;
+
+    if (newSearchQuery !== prevSearchQuery) {
+      // update value in input box
+      this.refs.searchInput.setState({value: newSearchQuery ? decodeURIComponent(newSearchQuery) : ``});
+      // update searchQuery state
+      this.setState({searchQuery: newSearchQuery});
     }
   },
   clearSearch() {
-    this.searchInput.value = ``;
-    this.searchInput.blur();
-  },
-  handleInputBlur() {
-    this.updateSearchQuery();
+    this.refs.searchInput.setState({value: ``}, () => {
+      this.updateSearchQuery();
+    });
   },
-  handleInputKeyUp(event) {
-    if (event.keyCode === 27) { // escape key
-      this.clearSearch();
-    }
+  handleInputChange() {
```

Comment:
If this doesn't do anything else, can you just call `updateSearchQuery` from the Debounce input itself?